### ISSUE TITLE
Slave selling giving thought to other maps

### ISSUE THREAD

**Issue by Mihokyo** [Vectorial1024/Desynchronized#6]
Hello

I don't know if it's related to the mod... 
I had a colonist who had no family or love relationship in my colony, she stayed about a year in it, then one day I arrest her to make her a slave and then sold her.   

But when I sold her, a bad thought happened on a male colon where its said "his wife would have been sold". He does have a wife but has not been sold since it is not the same person.  

Him didn't know who was the (colonist/slave)  
___________
**Thought:**  
[Loved one sold](https://rimworldwiki.com/wiki/Thoughts#Loved_one_sold)  
__________
**Pic:**  
<a href="https://www.noelshack.com/2019-13-5-1553891861-loved1.png"><img src="https://image.noelshack.com/minis/2019/13/5/1553891861-loved1.png" border="0" alt="1553891861-loved1.png - envoi d'image avec NoelShack" title="1553891861-loved1.png"/></a>

<a href="https://www.noelshack.com/2019-13-5-1553891861-loved2.png"><img src="https://image.noelshack.com/minis/2019/13/5/1553891861-loved2.png" border="0" alt="1553891861-loved2.png - envoi d'image avec NoelShack" title="1553891861-loved2.png"/></a>

**Comment by Vectorial1024** [comment]
I'm sorry that I am replying late.

Just to confirm, who were you trying to sell as a slave?

**Comment by Mihokyo** [comment]
Yes i sell someone as slave to another town

An ex-colon
I arrest and sell her to a city

I didn't sell Yoshiko but some one other.. she was Tanner, Davidus's mother

\+ They aren't in the same city. 
__________

This is a bad trad, in english version they didn't specify their name and who they are for Davidus.

But in french version the specify the name & the relation ship, like ("They sold my wife Hilary to a trader like a piece of meat!") 

____________
In conclusion this tought is not desyncronized :[Loved one sold]( https://rimworldwiki.com/wiki/Thoughts#Loved_one_sold)

**Comment by Vectorial1024** [comment]
OK, so Tanner is the mother of Davidus. When Tanner is sold away, Davidus is getting some thoughts.

1. A "Loved one sold" thought is given to Davidus. Yes, you are correct, this thought is not from my mod. It is from the base game. In the French translation, I can notice a place "{0}" that is used by the game to display the name Davidus.
2. Now, we look at the description of the thought. When I look at the source files, the French reads "Quelqu'un que j'aime a été vendu à un marchand, comme une vulgaire marchandise !". Since I can't read French, I used Google translate to translate the sentence into English. And from the result, no wife was mentioned. (Maybe you want to go to Ludeon's website and ask about the translations?)
3. Finally, we look at why Davidus received the thought. When I check my source code, I have found that I did not properly implement the thought-giving procedure of selling colonists away. This is a bug.

While there is some confusion going on, thanks for letting me find the bug!

**Comment by Mihokyo** [comment]
I have try to make the bug again without Desyncronized[v1.4.6] on a new game

I sell again some one with family link and they say a work like "Mon amoureux(se) Docter à été vendu"

I sell the father of Site (Named: Edouard Saito)

<a href="https://www.noelshack.com/2019-17-1-1555951560-familly.png"><img src="https://image.noelshack.com/minis/2019/17/1/1555951560-familly.png" border="0" alt="1555951560-familly.png - envoi d'image avec NoelShack" title="1555951560-familly.png"/></a>
<a href="https://www.noelshack.com/2019-17-1-1555951559-peopleincolony.png"><img src="https://image.noelshack.com/minis/2019/17/1/1555951559-peopleincolony.png" border="0" alt="1555951559-peopleincolony.png - envoi d'image avec NoelShack" title="1555951559-peopleincolony.png"/></a>
<a href="https://www.noelshack.com/2019-17-1-1555951565-thought.png"><img src="https://image.noelshack.com/minis/2019/17/1/1555951565-thought.png" border="0" alt="1555951565-thought.png - envoi d'image avec NoelShack" title="1555951565-thought.png"/></a>

The following word is:
"Mon amoureux(se) {0} à été vendu !"
"Quelqu'un que j'aime a été vendu à un marchand, comme une vulgaire marchandise !"

___________
**Conclusion, the translate bug isn't caused from you**

**Comment by Vectorial1024** [comment]
Glad to hear that!

Now this leaves only the map bug to fix.

**Comment by Vectorial1024** [comment]
The map bug is fixed in the latest update. (EDIT: v1.5.0)

Closing this for now since there are no outstanding issues here for now.
